shell_integration/windows/OCUtil/RemotePathChecker.cpp

```diff
@@ -172,10 +176,13 @@ RemotePathChecker::~RemotePathChecker()
     CloseHandle(_newQueries);
 }
 
-vector<wstring> RemotePathChecker::WatchedDirectories()
+void RemotePathChecker::UpdateWatchedDirectories(vector<wstring>& localWatchedDirectories, unsigned& localVersion) const
 {
+    if (localVersion == _watchedDirectoriesVersion)
```

Comment:
Access to "_watchedDirectoriesVersion"  without the lock!
Put it inside the lock or make it std::atomic